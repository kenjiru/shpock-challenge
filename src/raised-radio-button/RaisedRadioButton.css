.raised-radio-button {
    height: 30px;
}

.raised-radio-button input[type=radio] + div {
    align-items: center;
}