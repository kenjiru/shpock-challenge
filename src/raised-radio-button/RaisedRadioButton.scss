.raised-radio-button {
    height: 30px;

    input[type=radio] + div {
        align-items: center;
    }
}
