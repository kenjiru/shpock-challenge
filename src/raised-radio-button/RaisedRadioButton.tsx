import * as classNames from "classnames";
import * as React from "react";
import { Component, CSSProperties, ReactElement } from "react";
import { RadioButton, RadioButtonProps } from "material-ui";
import { grey700 } from "material-ui/styles/colors";

import RaisedContainer from "../raised-container/RaisedContainer";

import "./RaisedRadioButton.css";

class RaisedRadioButton extends Component<RadioButtonProps> {
    public static CLASS_NAME: string = "raised-radio-button";
    public static LINE_HEIGHT: string = "18px";
    public static CHECKED_COLOR: string = "white";
    public static NORMAL_COLOR: string = grey700;
    public static SIZE: number = 20;
    public static FONT_SIZE: number = 13;

    public render(): ReactElement<HTMLElement> {
        return (
            <RaisedContainer
                className={this.getClassName()}
                checked={this.props.checked}
            >
                <RadioButton
                    {...this.props}
                    iconStyle={this.getIconStyle()}
                    labelStyle={this.getLabelStyle()}
                />
            </RaisedContainer>
        );
    }

    private getIconStyle(): CSSProperties {
        const {checked} = this.props;

        return {
            fill: checked ? RaisedRadioButton.CHECKED_COLOR : RaisedRadioButton.NORMAL_COLOR,
            width: RaisedRadioButton.SIZE,
            height: RaisedRadioButton.SIZE
        };
    }

    private getLabelStyle(): CSSProperties {
        const {checked} = this.props;

        return {
            color: checked ? RaisedRadioButton.CHECKED_COLOR : RaisedRadioButton.NORMAL_COLOR,
            fontSize: RaisedRadioButton.FONT_SIZE,
            lineHeight: RaisedRadioButton.LINE_HEIGHT
        };
    }

    private getClassName(): string {
        return classNames(RaisedRadioButton.CLASS_NAME, this.props.className);
    }
}

export default RaisedRadioButton;
