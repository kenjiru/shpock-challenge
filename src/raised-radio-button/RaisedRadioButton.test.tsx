import * as React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import shallowToJson from "enzyme-to-json";
import { RadioButton, RadioButtonProps } from "material-ui";

import RaisedRadioButton from "./RaisedRadioButton";

describe("RaisedRadioButton", () => {
    it("should render correctly", () => {
        const output: ShallowWrapper = shallow(
            <RaisedRadioButton/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it("should pass the correct styles to the underlying RadioButton", () => {
        const output: ShallowWrapper = shallow(
            <RaisedRadioButton/>
        );

        let props: RadioButtonProps = output.find(RadioButton).props();

        expect(props.iconStyle.width).toBe(RaisedRadioButton.SIZE);
        expect(props.iconStyle.height).toBe(RaisedRadioButton.SIZE);
        expect(props.labelStyle.lineHeight).toBe(RaisedRadioButton.LINE_HEIGHT);
    });

    it("should have the correct color when it's checked", () => {
        const output: ShallowWrapper = shallow(
            <RaisedRadioButton checked={true}/>
        );

        let props: RadioButtonProps = output.find(RadioButton).props();

        expect(props.iconStyle.fill).toBe(RaisedRadioButton.CHECKED_COLOR);
        expect(props.labelStyle.color).toBe(RaisedRadioButton.CHECKED_COLOR);
    });

    it("should have the correct color when it's NOT checked", () => {
        const output: ShallowWrapper = shallow(
            <RaisedRadioButton checked={false}/>
        );

        let props: RadioButtonProps = output.find(RadioButton).props();

        expect(props.iconStyle.fill).toBe(RaisedRadioButton.NORMAL_COLOR);
        expect(props.labelStyle.color).toBe(RaisedRadioButton.NORMAL_COLOR);
    });
});
