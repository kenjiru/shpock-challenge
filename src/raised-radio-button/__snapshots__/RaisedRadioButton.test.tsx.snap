// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaisedRadioButton should render correctly 1`] = `
<MuiComponent
  className="raised-radio-button"
>
  <RadioButton
    checked={false}
    disabled={false}
    iconStyle={
      Object {
        "fill": "#616161",
        "height": 20,
        "width": 20,
      }
    }
    labelPosition="right"
    labelStyle={
      Object {
        "color": "#616161",
        "fontSize": 13,
        "lineHeight": "18px",
      }
    }
  />
</MuiComponent>
`;
